crates/ruff_linter/src/rules/pycodestyle/rules/blank_lines.rs

```diff
@@ -505,6 +517,15 @@ impl<'a> Iterator for LinePreprocessor<'a> {
                     // Set the start for the next logical line.
                     self.line_start = range.end();
 
+                    if self
+                        .cell_offsets
+                        .is_some_and(|cell_offsets| cell_offsets.contains(&self.line_start))
+                    {
+                        self.is_cell_first_non_comment_line = true;
+                    } else if !line_is_comment_only {
+                        self.is_cell_first_non_comment_line = false;
+                    }
+
```

Comment:
Is it okay that we set `is_cell_first_non_comment_line` to `true` even if `line_is_comment_only` is true?

Reply:
It's fine since if a cell starts with a comment we don't want to run the check on that comment (since it's the first line).
However the naming is pretty bad indeed.